src/test/scala/chisel3/tests/IntervalGcdTest.scala

```diff
@@ -0,0 +1,59 @@
+// See README.md for license details.
```

Comment:
Still not convinced this gives you much over the basic Interval interface test. Not a huge fan of more tests for the sake of more tests or the general scattershot approach of large, complex tests.

Reply:
The tests revealed things that I am not sure I would have found with a more spartan approach. I also would not underestimate the value of giving example usage of things in the tests suites. It might be better to create other venues for that but it is a lot of work. Whereas here the work is done in direct relation to functional testing.

Reply:
This might be a chisel cookbook example or something, but I'm not sure it belongs here. This is arguably more of how to build circuits in chisel3 w/ intervals than how to write tests in chiseltest. I'm sure there's a place for it somewhere (maybe "100 variations on GCD" or something) but I'm not sure that place is here...

Reply:
Yeah, this was a mistake. I think I have it deleted it now. Previous push failed somehow